completers/fastfetch_completer/cmd/root.go

```diff
@@ -0,0 +1,144 @@
+package cmd
+
+import (
+	"encoding/json"
+	"os/exec"
+	"strings"
+
+	"github.com/rsteube/carapace"
+	"github.com/spf13/cobra"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "fastfetch",
+	Short: "A neofetch-like tool for fetching system information and displaying them in a pretty way",
+	Long:  "https://github.com/fastfetch-cli/fastfetch",
+	Run:   func(cmd *cobra.Command, args []string) {},
+}
+
+func Execute() error {
+	return rootCmd.Execute()
+}
+
+type FlagData struct {
+	Short  string
+	Long   string
+	Desc   string
+	Remark string
+	Pseudo bool
+	Arg    *struct {
+		Type     string
+		Optional bool
+		Default  interface{}
+		Enum     map[string]string
+	}
+}
+
+func init() {
+	carapace.Gen(rootCmd).Standalone()
+
+	var groups map[string][]FlagData
+	if out, err := exec.Command("fastfetch", "--help-raw").Output(); err == nil {
```

Comment:
This (and the whole flag registration) needs to be done within [PreRun](https://rsteube.github.io/carapace/carapace/gen/preRun.html).
The develop build isn't optimized so it would be executed on every invocation of `carapace`.

Reply:
I tried to put the code into PreRun, and auto completion no longer worked.

Reply:
Put the `Standalone` in there as well. Seems some cobra defaults are interfering again.

```go
	carapace.Gen(rootCmd).PreRun(func(cmd *cobra.Command, args []string) {
		carapace.Gen(rootCmd).Standalone()
```